src/saltext/mysql/cache/mysql_cache.py

```diff
@@ -309,17 +309,23 @@ def flush(bank, key=None):
     cur.close()
 
 
-def ls(bank):
+def list_(bank):
     """
-    Return an iterable object containing all entries stored in the specified
-    bank.
+    Return an iterable object containing all entries stored in the specified bank.
     """
+    bank_like = bank + '%'
     _init_client()
-    query = "SELECT etcd_key FROM {} WHERE bank=%s".format(__context__["mysql_table_name"])
-    cur, _ = run_query(__context__.get("mysql_client"), query, args=(bank,))
+    banks_query = "SELECT bank FROM {} WHERE bank LIKE %s".format(
+        __context__["mysql_table_name"]
+    )
+    cur, _ = run_query(__context__.get("mysql_client"), banks_query, args=(bank_like,))
     out = [row[0] for row in cur.fetchall()]
     cur.close()
-    return out
+    data = []
```

Comment:
issue(blocking): This is does not deduplicate and is missing files. Take the following structure (key == etc_key for brevity):

```yaml
- bank: root/sub/dir
  key: foo
- bank: root/sub/dir
  key: bar
- bank: root/another_sub/dir
  key: baz
- bank: root
  key: file
```

Listing `root` should yield:

```yaml
- sub
- another_sub
- file
```

But this returns:

```
- sub
- sub
- another_sub
```

Please see the last paragraph of my [previous review comment](https://github.com/salt-extensions/saltext-mysql/pull/10#pullrequestreview-2411950755) for suggestions.

Reply:
Thank you for the review. I'm thinking of using the last part returned by mysql of the bank this should contain  ``sub, another_sub, file``.  Using sets sounds like a great idea to fix the duplicate entries. Does it happen the etcd_key that matches the full bank parameter, perhaps it could be fixed by one SQL query without LIKE. I really don't like the idea of having another SQL query.

I'm grateful for all that you have done, I will do some thinking of my own.

Reply:
It should work with a single query similar to this mockup (definitely needs adjustments, e.g. client initialization):

```py
def list_(bank):
    bank = bank.strip("/")
    query = "SELECT bank, etcd_key FROM {} WHERE bank LIKE %s".format(
        __context__["mysql_table_name"]
    )
    cur, _ = run_query(__context__.get("mysql_client"), keys_query, args=(bank,))
    res = set()
    for res_bank, res_key in cur.fetchall():
        if res_bank == bank:
            res.add(res_key)
        else:
            res.add(res_bank[len(bank) + 1:].split("/")[0])
    return list(res)
```